src/core/Akka.Remote/EndpointRegistry.cs

```diff
@@ -92,32 +104,38 @@ public IActorRef ReadOnlyEndpointFor(Address address)
 
         public bool IsWritable(IActorRef endpoint)
         {
-            return writableToAddress.ContainsKey(endpoint);
+            return _writableToAddress.ContainsKey(endpoint);
         }
 
         public bool IsReadOnly(IActorRef endpoint)
         {
-            return readonlyToAddress.ContainsKey(endpoint);
+            return _readonlyToAddress.ContainsKey(endpoint);
         }
 
         public bool IsQuarantined(Address address, int uid)
         {
-            var rvalue = false;
-            WritableEndpointWithPolicyFor(address).Match()
-                .With<EndpointManager.Quarantined>(q =>
-                {
-                    if (q.Uid == uid)
-                        rvalue = q.Deadline.HasTimeLeft;
-                })
-                .Default(msg => rvalue = false);
+            // timeOfRelease is only used for garbage collection. If an address is still probed, we should report the
+            // known fact that it is quarantined.
+            var policy = WritableEndpointWithPolicyFor(address) as EndpointManager.Quarantined;
+            return policy?.Uid == uid;
```

Comment:
for a remote node to still be "quarantined" the UID needs to match that of the incoming address